ports/esp32/machine_uart.c

```diff
@@ -465,10 +465,15 @@ STATIC mp_uint_t machine_uart_read(mp_obj_t self_in, void *buf_in, mp_uint_t siz
         time_to_wait = 0;
     } else {
         time_to_wait = pdMS_TO_TICKS(self->timeout);
+        MP_THREAD_GIL_EXIT();
```

Comment:
I think it should set a variable here, `release_gil = true`, and use that instead of the timeout variable.

Because, while the GIL is released another thread could run and change the timeout of the UART.

Reply:
(See `mp_hal_stdout_tx_strn()` for comparison.)

Reply:
Yes, that makes sense. I'll push an update.